moveit_core/collision_detection_fcl/include/moveit/collision_detection_fcl/collision_env_fcl.h

```diff
@@ -0,0 +1,162 @@
+
+  CollisionEnvFCL(const CollisionEnvFCL& other, const WorldPtr& world);
+
+  ~CollisionEnvFCL() override;
+
+  virtual void checkSelfCollision(const CollisionRequest& req, CollisionResult& res,
+                                  const robot_state::RobotState& state) const override;
+
+  virtual void checkSelfCollision(const CollisionRequest& req, CollisionResult& res,
+                                  const robot_state::RobotState& state, const AllowedCollisionMatrix& acm) const override;
+
+  virtual void checkRobotCollision(const CollisionRequest& req, CollisionResult& res,
+                                   const robot_state::RobotState& state) const override;
+
+  virtual void checkRobotCollision(const CollisionRequest& req, CollisionResult& res,
+                                   const robot_state::RobotState& state, const AllowedCollisionMatrix& acm) const override;
+
+  virtual void checkRobotCollision(const CollisionRequest& req, CollisionResult& res,
+                                   const robot_state::RobotState& state1, const robot_state::RobotState& state2,
+                                   const AllowedCollisionMatrix& acm) const override;
+
+  virtual void checkRobotCollision(const CollisionRequest& req, CollisionResult& res,
+                                   const robot_state::RobotState& state1,
+                                   const robot_state::RobotState& state2) const override;
+
+  virtual void checkWorldCollision(const CollisionRequest& req, CollisionResult& res) const override;
+
+  virtual void checkWorldCollision(const CollisionRequest& req, CollisionResult& res,
+                                   const AllowedCollisionMatrix& acm) const override;
+
+  virtual void distanceSelf(const DistanceRequest& req, DistanceResult& res,
+                            const robot_state::RobotState& state) const override;
+
+  virtual void distanceRobot(const DistanceRequest& req, DistanceResult& res, const robot_state::RobotState& state) const override;
+
+  void setWorld(const WorldPtr& world) override;
+
+protected:
+  /** \brief Updates the collision objects saved in the manager to reflect a new padding or scaling of the robot links
+   */
+  void updatedPaddingOrScaling(const std::vector<std::string>& links) override;
+
+  /** \brief Bundles the different checkSelfCollision functions into a single function */
+  void checkSelfCollisionHelper(const CollisionRequest& req, CollisionResult& res, const robot_state::RobotState& state,
+                                const AllowedCollisionMatrix* acm) const;
+
+  void checkRobotCollisionHelper(const CollisionRequest& req, CollisionResult& res,
+                                 const robot_state::RobotState& state, const AllowedCollisionMatrix* acm) const;
+
+  /** \brief Construct an FCL collision object from MoveIt's World::Object. */
+  void constructFCLObject(const World::Object* obj, FCLObject& fcl_obj) const;
```

Comment:
There are some method name collisions here, e.g., 2 `constructFCLObjects`. Be sure to differentiate which one is World and which one is Robot.

Reply:
I changed them to `constructFCLObjectRobot` and `constructFCLObjectWorld` respectively